File: src/jjoller/foxinabox/Action.java
package jjoller.foxinabox;

/**
 * Helper class for Action encoding
 */
public class Action {

    public static final int FOLD = -1;
    public static final int CHECK = 0;
    public static final int DEAL_PREFLOP = -2;
    public static final int DEAL_FLOP = -3;
    public static final int DEAL_TURN = -4;
    public static final int DEAL_RIVER = -5;

    public static String toString(int action) {

        switch (action) {
            case FOLD:
                return "f";
            case DEAL_PREFLOP:
                return "";
            case DEAL_FLOP:
                return "F";
            case DEAL_TURN:
                return "T";
            case DEAL_RIVER:
                return "R";
            default:
                return action+"";
        }
    }
}

File: src/jjoller/foxinabox/Action.kt
package jjoller.foxinabox

/**
 * Helper class for Action encoding
 */
object Action {

    val FOLD = -1
    val CHECK = 0
    val DEAL_PREFLOP = -2
    val DEAL_FLOP = -3
    val DEAL_TURN = -4
    val DEAL_RIVER = -5

    fun toString(action: Int): String {

        when (action) {
            FOLD -> return "f"
            DEAL_PREFLOP -> return ""
            DEAL_FLOP -> return "F"
            DEAL_TURN -> return "T"
            DEAL_RIVER -> return "R"
            else -> return action.toString();
        }
    }
}
